update_copyrights

```diff
@@ -97,37 +97,46 @@ with open(options.input, encoding="utf-8") as csvfile:
-            file_hash = do_hash(file_path)
-
-            if not file_path in csv_data:
-                added.append(["", file_path, "", "", "", do_git(file_path), file_hash])
-            elif len(csv_data[file_path]) < 7:
-                missing_fields.append(csv_data[file_path])
-            elif len(csv_data[file_path]) > 7:
-                extra_fields.append(csv_data[file_path])
-            elif csv_data[file_path][5] != "":
-                update.append(csv_data[file_path])
-            elif csv_data[file_path][6] != file_hash:
-                csv_data[file_path][5] = do_git(file_path)
-                csv_data[file_path][6] = file_hash
-                changed.append(csv_data[file_path])
-            elif csv_data[file_path][2].strip() == "" or csv_data[file_path][3].strip() == "":
-                incomplete.append(csv_data[file_path])
-            elif not csv_data[file_path][2] in known_licenses:
-                unknown_licenses.append(csv_data[file_path][2])
-                incomplete.append(csv_data[file_path])
-            else:
-                unchanged.append(csv_data[file_path])
+try:
+    for root, _, files in os.walk(options.repo):
+        for filename in files:
+            filetype = Path(filename).suffix
+            if filetype == ".png" or filetype == ".jpg" or filetype == ".webp" or filetype == ".wav" or filetype == ".ogg":
+                file_path = os.path.normpath(os.path.join(root, filename))
+                if os.path.sep != '/':
+                    # Always use slashes for the file path irrespective of OS used to run the update
+                    file_path = file_path.replace(os.path.sep, '/')
+                file_hash = do_hash(file_path)
+
+                if not file_path in csv_data:
+                    added.append(["", file_path, "", "", "", do_git(file_path), file_hash])
+                elif len(csv_data[file_path]) < 7:
+                    missing_fields.append(csv_data[file_path])
+                elif len(csv_data[file_path]) > 7:
+                    extra_fields.append(csv_data[file_path])
+                elif csv_data[file_path][5] != "":
+                    update.append(csv_data[file_path])
+                elif csv_data[file_path][6] != file_hash:
+                    csv_data[file_path][5] = do_git(file_path)
+                    csv_data[file_path][6] = file_hash
+                    changed.append(csv_data[file_path])
+                elif csv_data[file_path][2].strip() == "" or csv_data[file_path][3].strip() == "":
+                    incomplete.append(csv_data[file_path])
+                elif not csv_data[file_path][2] in known_licenses:
+                    unknown_licenses.append(csv_data[file_path][2])
+                    incomplete.append(csv_data[file_path])
+                else:
+                    unchanged.append(csv_data[file_path])
+    if options.input == options.output:
+        os.remove(options.input + '-bak')
+except:
```

Comment:
Are there certain types of errors you're wanting to catch here? I think, generally, it's not a good idea to catch all errors.

Reply:
When the error is rethrown it is ok to catch everything